WordPress/src/main/java/org/wordpress/android/ui/people/RoleChangeDialogFragment.java

```diff
@@ -0,0 +1,160 @@
+    private static final String ROLE_TAG = "role";
+
+    private RoleListAdapter mRoleListAdapter;
+    private OnChangeListener mListener;
+
+    public static RoleChangeDialogFragment newInstance(long personID, int localTableBlogId, String role) {
+        RoleChangeDialogFragment roleChangeDialogFragment = new RoleChangeDialogFragment();
+        Bundle args = new Bundle();
+
+        args.putLong(PERSON_ID_TAG, personID);
+        args.putInt(PERSON_LOCAL_TABLE_BLOG_ID_TAG, localTableBlogId);
+        if (role != null) {
+            args.putString(ROLE_TAG, role);
+        }
+
+        roleChangeDialogFragment.setArguments(args);
+        return roleChangeDialogFragment;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        try {
+            mListener = (OnChangeListener) context;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(context.toString() + " must implement OnChangeListener");
+        }
+    }
+
+    // We need to override this for devices pre API 23
+    @SuppressWarnings("deprecation")
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        try {
+            mListener = (OnChangeListener) activity;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString() + " must implement OnChangeListener");
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+    }
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity(), R.style.Calypso_AlertDialog);
+        builder.setTitle(R.string.role);
+        builder.setNegativeButton(R.string.cancel, null);
+        builder.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                if (mListener != null) {
+                    String role = mRoleListAdapter.getSelectedRole();
+                    Bundle args = getArguments();
+                    if (args != null) {
+                        long personID = args.getLong(PERSON_ID_TAG);
+                        int localTableBlogId = args.getInt(PERSON_LOCAL_TABLE_BLOG_ID_TAG);
+                        mListener.onRoleChanged(personID, localTableBlogId, role);
+                    }
+                }
+            }
+        });
+
+        if (mRoleListAdapter == null) {
+            final String[] roles = getResources().getStringArray(R.array.roles);
+            mRoleListAdapter = new RoleListAdapter(getActivity(), R.layout.role_list_row, roles);
+        }
+        Bundle args = getArguments();
+        if (args != null) {
+            String role = args.getString(ROLE_TAG);
+            mRoleListAdapter.setSelectedRole(role);
```

Comment:
Let's also cover the case where the device is rotating with the dialog open. In the usual drill, we can use the `savedInstanceState` to pass the currently selected Role. Cheers!